### PR TITLE
[nunjucks] Fix incorrect loader extend type in type definition

### PR DESCRIPTION
**Pull request by JemarJones**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://mozilla.github.io/nunjucks/api.html#writing-a-loader https://github.com/mozilla/nunjucks/blob/d34fdbff8509327db7ab15b3672b3005b89a229a/nunjucks/src/object.js#L76 https://github.com/mozilla/nunjucks/blob/d34fdbff8509327db7ab15b3672b3005b89a229a/nunjucks/src/loader.js#L6
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Regarding the last checkbox above: There are additional types added in the newer version of nunjucks which i haven't added here, so i suppose this change doesn't bring it up to date with the newest version? 

But further than this i have a bigger concern about the versioning here. There are also the previously defined extend types which are incorrect:
https://github.com/JemarJones/DefinitelyTyped/blob/fb86becaa2f362b8380f7139c3cb0830c5bd8c59/types/nunjucks/index.d.ts#L117
https://github.com/JemarJones/DefinitelyTyped/blob/fb86becaa2f362b8380f7139c3cb0830c5bd8c59/types/nunjucks/index.d.ts#L107

^ These are definitions for an instance method which does not exist (its static, as my new types show). As such if you try to call `extend` on an instance of the Loader class as the existing type definition defines it, you will just get a runtime error because `extend` is not defined. I think it would be best to remove those incorrect types, but i'm not quite sure what the procedure would be in terms of versioning. Removing those would technically be a breaking change, but the code that it breaks would just be failing at runtime currently. I hope someone can advise what i should do if i am to remove the incorrect type. The DefinetelyTyped [documentation](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library) on versioning is mum on the issue of a breaking change that doesn't correspond to a new major version for the typed package. For now i haven't removed the incorrect type from the loader class, i've only added the correct type for the static version of the method that actually does exist.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## nunjucks ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nunjucks@latest/))
was missing the following properties:
1. compiler
1. parser
1. lexer
1. nodes
1. reset
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fb86becaa2f362b8380f7139c3cb0830c5bd8c59
</p>

**Comment by RubenSlabbert** [comment]
I would appreciate a member of the DefinitelyTyped team's opinion on this, but my personal preference is that since the types as-is are incorrect, and the version of the types is intended to correspond to the version of the library, this change should be pushed as a patch version on the current types version. As to your point around runtime errors, I wouldn't consider this a "breaking change" as such since it's already broken, rather I would consider it a bug fix.

Given the above, I am happy for the change to be approved with a patch bump.

File: types/nunjucks/index.d.ts [hunk 1]
```diff
@@ -114,7 +114,8 @@ export class Loader {
     emit(name: string, ...args: any[]): void;
     resolve(from: string, to: string): string;
     isRelative(filename: string): boolean;
-    extend(toExtend: ILoader): ILoader;
+    extend(extender: ILoader): ILoader;
```
Comment (RubenSlabbert) [inline types/nunjucks/index.d.ts:117]: To your point, I believe this line should be removed to bring it in line with runtime behaviour?

**Review by rslabbert — APPROVED** [review]

File: types/nunjucks/index.d.ts (hunk 1 above)
Comment (JemarJones) [reply types/nunjucks/index.d.ts]: Sounds good, updated 👍

**Review by rslabbert — APPROVED** [review]

**Comment by JemarJones** [comment]
Ready to merge

**Pull request merged**